src/Microsoft.NetCore.Analyzers/Core/Security/JsonNetTypeNameHandling.cs

```diff
@@ -94,7 +93,7 @@ public override void Initialize(AnalysisContext analysisContext)
 
                     bool IsOtherThanNone(IFieldReferenceOperation fieldReferenceOperation)
                     {
-                        if (!typeNameHandlingSymbol.Equals(fieldReferenceOperation.Field.ContainingType))
+                        if (!typeNameHandlingSymbol!.Equals(fieldReferenceOperation.Field.ContainingType))
```

Comment:
>! [](start = 51, length = 1)

Don't think this one is necessary.

Reply:
Unfortunately the compiler is not performing interprocedural analysis of local functions, so it assumes this can be null. I will add `RoslynDebug.Assert(typeNameHandlingSymbol != null);` before this statement to avoid the suppression with `!` operator.